---
title: "review"
author: "chiteri"
date: "3/21/2021"
output: html_document
---

# Chiteri's review
#Great work on this assignment Maggie!
#I only have a few comments as suggested within the document



## R Homework Code 
Step 1: Get Data Into R
```{r}
library(tidyverse)
SNPData <- read_tsv("https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/snp_position.txt")
FangData <- read_tsv("https://raw.githubusercontent.com/EEOB-BioData/BCB546-Spring2021/main/assignments/UNIX_Assignment/fang_et_al_genotypes.txt")
```
	Step 2: Data Inspection
```{r}
  ncol(SNPData)
  ncol(FangData)
  nrow(SNPData)
  nrow(FangData)
	typeof(SNPData)
	typeof(FangData)
	
	#Comment out this two since the dataframes have so many columns
	
	#names(SNPData) 
	#names(FangData)
```
Step 3: Data Processing 
*Separate Maize and Teosinte Data In Fang File and Head to check them*
```{r}
FangData_Maize <- FangData[which(FangData$Group=="ZMMIL" | FangData$Group =="ZMMLR" | FangData$Group == "ZMMMR"),]
	head(FangData_Maize)
	
		FangData_Teosinte <-FangData[which(FangData$Group=="ZMPBA" | FangData$Group =="ZMPIL" | FangData$Group == "ZMPJA"),]
	head(FangData_Teosinte)
```
*Prep SNP data*
```{r}
#Select snp_id, chromosome number and snp position
SNPData_SpecificC <- SNPData[,c(1,3,4)]
```
*Transpose Data*
```{r}
MaizeData_transposed <- as.data.frame(t(FangData_Maize))
TeosinteData_transposed <- as.data.frame(t(FangData_Teosinte))
```
*Trim Rows and Name Column*
```{r}
	colnames(MaizeData_transposed) <- as.character(unlist(MaizeData_transposed[1,])) 
	colnames(TeosinteData_transposed) <- as.character(unlist(TeosinteData_transposed[1,]))
	MaizeData_trim <- MaizeData_transposed[-c(1:3),]
	TeosinteData_trim <- TeosinteData_transposed[-c(1:3),]
```
*Join two datafiles, SNPData_Specific and Fang Data, teosinte data*
```{r}
MaizeData_joined <- merge(x = SNPData_SpecificC, y = MaizeData_trim, by.x = "SNP_ID", by.y ="row.names", all.y = TRUE)

TeosinteData_joined <- merge(x = SNPData_SpecificC, y = TeosinteData_trim, by.x = "SNP_ID", by.y ="row.names", all.y = TRUE)
```
*Make SNP Increasing*
```{r}
#If you could filter out the unknown and multiple files. it would be great

#maiz_filtered <- filter(MaizeData_joined, Chromosome != 'unknown' | Chromosome != 'multiple')

	MaizeData_joined <- MaizeData_joined[order(MaizeData_joined$Position),] 
	TeosinteData_joined <- TeosinteData_joined[order(TeosinteData_joined$Position),]
```

*Put Each increase chromosome in its own .csv File*
```{r}
#Great loop
	for (a in 1:10){
  		MaizeData_Chromosome_Ordered <- MaizeData_joined[MaizeData_joined$Chromosome == a, ]
  		write.csv(MaizeData_Chromosome_Ordered, file= paste("Maize_chromosome", a, ".csv", sep=""), row.names = F)
		}
	for (a in 1:10){
  		TeosinteData_Chromosome_Ordered <- TeosinteData_joined[TeosinteData_joined$Chromosome == a, ]
  		write.csv(TeosinteData_Chromosome_Ordered, file= paste("Teosinte_chromosome", a, ".csv", sep=""), row.names = F)
		}
```
*Make SNP Decreasing*
```{r}
MaizeData_joined_decrease <- MaizeData_joined[order(MaizeData_joined$Position, decreasing=T),]

TeosinteData_joined_decrease <- TeosinteData_joined[order(TeosinteData_joined$Position, decreasing=T),]
```
*Put Each decrease chromosome in its own .csv File*
```{r}
for (a in 1:10){
 		MaizeData_DecreaseOrdered<-MaizeData_joined_decrease[MaizeData_joined_decrease$Chromosome == a, ]
  		write.csv(MaizeData_joined_decrease, file= paste("Maize_chromosome", a, "d.csv", sep=""), row.names = F)
		}
	for (a in 1:10){
 		TeosinteData_DecreaseOrdered<-TeosinteData_joined_decrease[TeosinteData_joined_decrease$Chromosome == a, ]
  		write.csv(TeosinteData_joined_decrease, file= paste("Teosinte_chromosome", a, "d.csv", sep=""), row.names = F)
	}

#I noticed you were not able to complete a part of part of data processing.
#Replacing ? with -
#You could use this code to help you out

#find "?" and replace "-", 
#maize_data_replaced <- as_tibble(lapply(maize_data, gsub, pattern="?", replacement="-", fixed=TRUE))
```
Step 4: Data Visualization 
*Package Installation/Setup*
```{r}
#It is good to load all your packages at the start of the script
#I think, you don't need to transpose the data here again. You can use the previous transposed dataframe
library(ggplot2)
library(tidyverse)
library(reshape2)
library(dplyr)
library(plyr)

FangData_transposed2 <- as.data.frame(t(FangData))
colnames(FangData_transposed2) <- as.character(unlist(FangData_transposed2[1,]))
FangData_SNPData_joined <- merge(x = SNPData_SpecificC, y = FangData_transposed2, by.x = "SNP_ID", by.y ="row.names", all.y = TRUE)
```

##SNP/Chromosome
```{r}
#To put chromosomes in order
FangData_SNPData_joined$Chromosome = factor(FangData_SNPData_joined$Chromosome, levels = c(1,2,3,4,5,6,7,8,9,10))

ggplot(FangData_SNPData_joined, mapping = aes((Chromosome))) + geom_bar() + ggtitle("SNPs Per Chromosome") + labs(x="Chromosome",y="Total SNP Number")
```
Missing data and amount of Heterozygosity
```{r}
headers <- colnames(FangData)[-c(1:3)]
FangData_melted <- melt(FangData, measure.vars = headers)

FangData_melted[ FangData_melted == "?/?" ] = NA
FangData_melted$isHomozygous <- (FangData_melted$value=="A/A" | FangData_melted$value=="C/C" | FangData_melted$value=="G/G" | FangData_melted$value=="T/T")


```
*Ordered*
```{r}
FangData_melted_bysample <- FangData_melted[order(FangData_melted$Sample_ID),]
```
*Counted*
```{r}
FangData_bysample_count <- ddply(FangData_melted_bysample, c("Sample_ID"), summarise, counting_homozygous=sum(isHomozygous, na.rm=TRUE), counting_heterozygous=sum(!isHomozygous, na.rm=TRUE), isNA=sum(is.na(isHomozygous)))

FangData_bysample_count_melted <- melt(FangData_bysample_count, measure.vars = c("counting_homozygous", "counting_heterozygous", "isNA"))
```
*Order Better for Plotting*
```{r}
FangData_grouped <- FangData_melted[order(FangData_melted$Group),]
FangData_grouped_counted <- ddply(FangData_grouped, c("Group"), summarise, counting_homozygous=sum(isHomozygous, na.rm=TRUE), counting_heterozygous=sum(!isHomozygous, na.rm=TRUE), isNA=sum(is.na(isHomozygous)))
FangData_grouped_counted_melted <- melt(FangData_grouped_counted, measure.vars = c("counting_homozygous", "counting_heterozygous", "isNA"))
```
*Plotting2*
```{r}
ggplot(FangData_grouped_counted_melted,aes(x = Group, y= value, fill=variable)) + geom_bar(stat = "identity", position = "stack")
```
Your own visualization- Plot of how many of each group of SNP is contained in FangData
```{r}
ggplot(data = FangData_grouped) + geom_density(mapping = aes(x=value), fill="blue")
```

	
	


